cpp/CMakeLists.txt

```diff
@@ -57,6 +57,7 @@ option(BUILD_C_LIBRARY "Build cuVS C API library" OFF)
 option(BUILD_C_TESTS "Build cuVS C API tests" OFF)
 option(BUILD_CUVS_BENCH "Build cuVS ann benchmarks" OFF)
 option(BUILD_CAGRA_HNSWLIB "Build CAGRA+hnswlib interface" ON)
+option(BUILD_MG_ALGOS "Build with multi-GPU support" ON)
```

Comment:
Summarizing our offline conversation here (somewhere threaded, that could be marked `Resolved` and collapsed).

I just pushed https://github.com/rapidsai/cuvs/pull/231/commits/55fbb3646fb2e85cf0e01631ebd15ea1efeab536, which does the following:

* makes multi-GPU support on by default for wheel and conda builds
* adds a `build.sh` option, `--no-mg`, to run a build with it not enabled
* adds a CMake option here, to make the default setting here in CMake explicit (so that, for example, wheels will be built with this support enabled by default)